service/history/api/respondworkflowtaskcompleted/workflow_task_completed_handler.go

```diff
@@ -423,6 +425,13 @@ func (handler *workflowTaskCompletedHandler) handleCommandScheduleActivity(
 	executionInfo := handler.mutableState.GetExecutionInfo()
 	namespaceID := namespace.ID(executionInfo.NamespaceId)
 
+	// TODO(fairness): remove this again once the SDK allows setting the fairness key
+	fairnessKeyPrefix := "x-temporal-internal-fairness-key-"
+	if strings.HasPrefix(attr.GetActivityId(), fairnessKeyPrefix) {
+		attr.Priority = cmp.Or(attr.Priority, &commonpb.Priority{})
+		attr.Priority.FairnessKey = strings.TrimPrefix(attr.GetActivityId(), fairnessKeyPrefix)
+	}
```

Comment:
there's a new CutPrefix:
```suggestion
	const fairnessKeyPrefix = "x-temporal-internal-fairness-key-"
	if key, ok := strings.CutPrefix(attr.GetActivityId(), fairnessKeyPrefix); ok {
		attr.Priority = cmp.Or(attr.Priority, &commonpb.Priority{})
		attr.Priority.FairnessKey = key
	}
```

Reply:
I'm not quite sure this is enough though since don't activity ids have to be unique, so we couldn't use the same fairness key twice? or at least twice before the previous one completes?

maybe we need to scan for a delimiter in the middle instead of a prefix?